stages/org.osbuild.shell.init

```diff
@@ -27,7 +27,10 @@ SCHEMA_2 = r"""
               "additionalProperties": false,
               "required": ["key", "value"],
               "properties": {
-                "key": "string",
+                "key": {
+                  "type": "string",
+                  "pattern": "[a-zA-Z][a-zA-Z0-9_]*"
```

Comment:
The document that you linked says 
```text
Environment variable names used by the utilities in the Shell and Utilities volume of
POSIX.1-2017 consist solely of uppercase letters, digits, and the <underscore> ( '_' )
from the characters defined in [Portable Character Set]
(https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap06.html#tagtcjh_3)
and do not begin with a digit.
```
So no lowercase letters... I'm not sure what `The name space of environment variable names containing lowercase letters is reserved for applications.` means in the context of this stage 🤔

Reply:
I feel it's technically allowed but might not be strictly POSIX for global env.  I'm good with restricting to ALL CAPS if you think it's more correct.

Reply:
Unless we already have a requirement to allow lowercase (e.g. from the EAP team), I would stick to all caps. We can always relax it... I'm also OK with keeping it as is if others are also OK with it

Reply:
I'll go with all caps.

Reply:
All caps now.  Updated tests.  I also added `_` as an allowed character in the first set, which is valid.
I tried to get the test to fail to see if it all works but realised that the pattern doesn't affect the validation.  Is that right?  Did I write the schema wrong?

Reply:
@achilleas-k Are you matching against the whole string or just a partial match? For full match you'll probably need to add anchors `^[A-Z_][A-Z0-9_]*$`

Reply:
Same.  Just ran a build now.
```
❯ grep pattern /usr/lib/osbuild/stages/org.osbuild.shell.init
    "patternProperties": {
                  "pattern": "^[A-Z_][A-Z0-9_]*$"

❯ cat ~/tmp/tree/etc/profile.d/test_env
export -VAR="val"
```

Reply:
https://github.com/osbuild/osbuild/commit/690a8fc42af46b700b60314b0fc84cd09da96d16

Reply:
That might be something specific to object keys though.

Reply:
Thanks for the help!